src/main/java/com/aws/greengrass/builtin/services/configstore/ConfigStoreIPCEventStreamAgent.java

```diff
@@ -270,7 +274,14 @@ public UpdateConfigurationResponse handleRequest(UpdateConfigurationRequest requ
 
                 Object value = request.getValueToMerge();
 
-                Topics serviceTopics = kernel.findServiceTopic(serviceName);
+                logger.atInfo().kv(SERVICE_NAME, serviceName).log(String.format(
+                        "Updating configuration for target component %s, key path %s, and value keys {%s}",
+                        targetServiceName, //todo-krickar fix value replacement
+                        Arrays.toString(keyPath),
+                        String.join(", ", request.getValueToMerge().keySet())
+                ));
+
+                Topics serviceTopics = kernel.findServiceTopic(targetServiceName);
```

Comment:
you need to check for authorization before starting to do the lookup and applying changes.

Reply:
Will do